src/core/model_accessing_API.jl

```diff
@@ -27,6 +29,14 @@ function add_agent_to_model!(agent::AbstractAgent, model::VecABM)
     return
 end
 
+function add_agent_to_model!(agent::AbstractAgent, model::StructABM)
+    agent.id != nagents(model) + 1 && error(lazy"Cannot add agent of ID $(agent.id) in a vector ABM of $(nagents(model)) agents. Expected ID == $(nagents(model)+1).")
+    tuple_agent = NamedTuple(x => getfield(agent, x) for x in fieldnames(typeof(agent)))
+    push!(agent_container(model), tuple_agent)
```

Comment:
I have the feeling that pushing is slow this way, worth benchmarking

Reply:
If this is so, to solve this, one should go down the calling path, and have a specialized version of `add_agent!` which doesn't construct the agent (if this is not already passed, and instead only its argument are passed)

Reply:
but this last thing is clearly advantageous in any case